engine/src/main/java/org/destinationsol/game/screens/ShipMixedControl.java

```diff
@@ -95,6 +99,33 @@ public void update(SolApplication solApplication, boolean enabled) {
         }
     }
 
+    /**
+     * When alt+tabbing in full screen mode, the client area can become 0 with/height, and when alt+tabbing back,
+     * the camera matrix can contain NaN values for the first frame
+     * @return true if projections should not be done this update
+     */
+    private boolean problemWithProjections(SolCam camera) {
+        return Gdx.graphics.getWidth() == 0 || Gdx.graphics.getHeight() == 0 ||
+                camera.getViewWidth() <= 0.f || camera.getViewHeight() <= 0.f || !camera.isMatrixValid();
+    }
+
+    /**
+     * Assert that the following vectors do not contain NaN or infinite values
+     * @param mouseCoords mouse coordinates in [0;1] screen space
+     * @param heroCoords hero coordinates projected to [0;1] screen space
+     */
+    private void assertHeroAndMouseCoords(Vector2 mouseCoords, Vector2 heroCoords) {
+        if(Double.isNaN(mouseCoords.x) || Double.isNaN(mouseCoords.y)) {
+            System.err.println("Screen size: " + Gdx.graphics.getWidth() + " " + Gdx.graphics.getHeight());
+            throw new RuntimeException("Mouse coordinates are not valid: " + mouseCoords.x + " " + mouseCoords.y);
+        }
+
+        if(Double.isNaN(heroCoords.x) || Double.isNaN(heroCoords.y)) {
+            System.err.println("Screen size: " + Gdx.graphics.getWidth() + " " + Gdx.graphics.getHeight());
```

Comment:
Logger

Reply:
That was a debug print which is not longer needed, so I removed it in the next commit